src/hash.hpp

```diff
@@ -8,10 +8,54 @@
 #define HASH_HPP_INCLUDED
 
 #include <string>
+#include "util/secure_string.hpp"
+
+enum HashFunc
+{
+    NONE = 0,
+    SHA256 = 1,
+    BCRYPT = 2
+};
+
+class Hasher
+{
+public:
+    HashFunc hashFunc;
```

Comment:
My understanding of const in c++ is a little fuzzy. If I can make this immutable that would be ideal (kind of like a C# get-only property). Otherwise I'll change it to be private.